staging/src/k8s.io/apiserver/pkg/storage/storagebackend/config.go

```diff
@@ -50,6 +50,9 @@ type Config struct {
 	Copier runtime.ObjectCopier
 	// Transformer allows the value to be transformed prior to persisting into etcd.
 	Transformer value.Transformer
+
+	// DoCompaction is a flag to turn on or off requesting compaction from apiserver.
+	DoCompaction bool
```

Comment:
I'm fine with exposing this as an option - I've already heard about customers who would like to control it themselves too.

I'm not sure if DoCompaction is the best name though.
I think that "auto-compaction" should be default and you should be able to opt-out. That would mean I would suggest changing to:
```
DisableCompaction
```
[with default false].

Reply:
Thanks for your comment. I'll rename the flag to `DisableCompaction`.

Reply:
@wojtek-t BTW can I add `DisableCompaction` to `EtcdOptions` instead of `storagebackend.Config`? It seems that we don't have a way of configuring `storagebackend.Config` by command line option or other ways.

Reply:
Yeah - that is fine

Reply:
@wojtek-t sorry I found a very simple way to do this... please ignore the above comment